### PR TITLE
Add Maintainers Guide and Release Guide to OSB

### PR DESCRIPTION
**Pull request by IanHoang**
### Description
This PR adds a maintainers guide and an updated release guide.

### Testing
N/A

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: MAINTAINERS_GUIDE.md
Comment (gkamat) [inline MAINTAINERS_GUIDE.md:7]: Maintainers can select any issues they would like to work on and should add should add them to the `In Progress` column of the [roadmap board](https://github.com/opensearch-project/opensearch-benchmark/projects/1).  Any issue added to the `In Progress` column should be properly labeled. For more information on how to properly label issues and PRs, see the [labels](#labels) section of this document.

Maintainers should create new issues as needed.

File: MAINTAINERS_GUIDE.md
Comment (gkamat) [inline MAINTAINERS_GUIDE.md:10]: and prioritizing them based on the [roadmap board](https://github.com/opensearch-project/opensearch-benchmark/projects/1).

File: MAINTAINERS_GUIDE.md
Comment (gkamat) [inline MAINTAINERS_GUIDE.md:14]: should regularly review the backlog of pull requests.

File: MAINTAINERS_GUIDE.md
Comment (gkamat) [inline MAINTAINERS_GUIDE.md:17]: on the release process, see

File: MAINTAINERS_GUIDE.md
Comment (gkamat) [inline MAINTAINERS_GUIDE.md:22]: Issues, pull requests and releases may be tagged with labels to categorize them.  Here are some suggestions on how to use labels.

File: MAINTAINERS_GUIDE.md
Comment (gkamat) [inline MAINTAINERS_GUIDE.md:24]: should be assigned to a selected subset of issues and not all issues.

File: MAINTAINERS_GUIDE.md
Comment (gkamat) [inline MAINTAINERS_GUIDE.md:31]: **Release Labels:**  Releases are tagged with labels in the scheme `vN.N.N`, for instance `v1.0.0`, ...

File: MAINTAINERS_GUIDE.md
Comment (gkamat) [inline MAINTAINERS_GUIDE.md:33]: in this open-source project and is run by a release manager volunteering from amongst the [maintainers](MAINTAINERS.md).

File: RELEASE_GUIDE.md
Comment (gkamat) [inline RELEASE_GUIDE.md:2]: ## Table of Contents

File: RELEASE_GUIDE.md
Comment (gkamat) [inline RELEASE_GUIDE.md:13]: this project.

File: RELEASE_GUIDE.md
Comment (gkamat) [inline RELEASE_GUIDE.md:19]: `opensearch-benchmark` projects maintain

File: RELEASE_GUIDE.md
Comment (gkamat) [inline RELEASE_GUIDE.md:24]: cherry-picking changes and opening a PR

File: RELEASE_GUIDE.md
Comment (gkamat) [inline RELEASE_GUIDE.md:28]: with the exception 
Name feature branches `feature/<FEATURE>`.

Once the feature branch has been merged into `main`, make sure to delete it.

File: RELEASE_GUIDE.md
Comment (gkamat) [inline RELEASE_GUIDE.md:32]: Same comment as from the other file above applies here as well.

File: RELEASE_GUIDE.md
Comment (gkamat) [inline RELEASE_GUIDE.md:34]: in this project
from amongst the maintainers

**Review by gkamat — CHANGES_REQUESTED** [review]

File: RELEASE_GUIDE.md
Comment (gkamat) [inline RELEASE_GUIDE.md:112]: Delete the ...

File: RELEASE_GUIDE.md
Comment (gkamat) [inline RELEASE_GUIDE.md:43]: that `version.txt` matches (perhaps global search-and-replace)
For example, if OSB is currently at version ..

File: RELEASE_GUIDE.md
Comment (gkamat) [inline RELEASE_GUIDE.md:44]: you have cloned the official OpenSearch Benchmark git repository.

Does it matter if it is the local computer?  This could be done on an EC2 instance.

File: RELEASE_GUIDE.md
Comment (gkamat) [inline RELEASE_GUIDE.md:45]: the documention.

File: RELEASE_GUIDE.md
Comment (gkamat) [inline RELEASE_GUIDE.md:47]: the new version

File: RELEASE_GUIDE.md
Comment (gkamat) [inline RELEASE_GUIDE.md:56]: an automated issue
the Jenkins workflow

File: RELEASE_GUIDE.md
Comment (gkamat) [inline RELEASE_GUIDE.md:57]: Once the maintainer has commented,
uploads OSB to the PyPI and OSB..
the person who pushed the tag
DockerHub staging to verify that the artifacts have been uploaded.

File: RELEASE_GUIDE.md
Comment (gkamat) [inline RELEASE_GUIDE.md:58]: Check the progress
in the Jenkins console

File: RELEASE_GUIDE.md
Comment (gkamat) [inline RELEASE_GUIDE.md:73]: the image
PyPi -> PyPI (global search-and-replace)

**Review by gkamat — CHANGES_REQUESTED** [review]

**Review by gkamat — APPROVED** [review]

**Review by beaioun — APPROVED** [review]
LGTM!

**Pull request merged**
